app/modelsbase.py

```diff
@@ -157,6 +158,10 @@ def update_from_json(self, j):
                     # {'0': {'a': 'b'}, '1': {'a': 'c'}}
                     items = [(int(i), v) for i, v in dict(subj).items()]
                     from_json = [c.from_json(v) for _, v in sorted(items)]
+                elif all(isinstance(uuid.UUID(_), uuid.UUID) for _ in subj.keys()):
```

Comment:
I think its really dangerous modifying this code - I would refactor the case study code we changed